### DISCUSSION TITLE
Can `showInTaskbar` be changed at runtime with TwinBasic?

### DISCUSSION THREAD

**Discussion by roke-julian-lockhart** [twinbasic/twinbasic#1630] [Q&A]
It's a limitation that stopped me using VB6 a little while ago, and I'm uncertain how I'd test it now because I lost what familiarity I had with VB6. Apologies if it's a stupid question.

https://github.com/yereverluvinunclebert/Panzer-Earth-gauge-VB6/issues/3#issuecomment-1642183600 made me remember it, and since I'm having issues with [`bugs.winehq.org/show_bug.cgi?id=55281#c0`](https://bugs.winehq.org/show_bug.cgi?id=55281#c0:~:text=Log%20out%20gugutyg4@rokejulianlockhart.addy.io-,%60showInTaskbar%60%20Win32%20API%20is%20unsupported.,-Bug%20List:%20\(4) I thought I'd ask here whether this is natively supported.

**Comment by fafalone** [comment]
Not currently, but your comment suggests other languages allow this, as does a brief search suggesting it's just a matter of toggling the `WS_EX_APPWINDOW` style, in which case it would be a fairly minor feature to add, so there's a good chance Wayne is open to it.

Does [this](https://www.vbforums.com/showthread.php?357131-Set-Get-the-ShowInTaskbar-property-for-a-window-at-runtime) work--

```
Private Declare Function GetWindowLong Lib "user32" Alias "GetWindowLongA" _
    (ByVal hwnd As Long, ByVal nIndex As Long) As Long
    
Private Declare Function SetWindowLong Lib "user32" Alias "SetWindowLongA" _
    (ByVal hwnd As Long, ByVal nIndex As Long, ByVal dwNewLong As Long) As Long
 
Private Declare Function ShowWindow Lib "user32" _
    (ByVal hwnd As Long, ByVal nCmdShow As Long) As Long
    
Private Const GWL_EXSTYLE = (-20)
Private Const WS_EX_APPWINDOW = &H40000
 
Const SW_HIDE = 0
Const SW_NORMAL = 1
 
Private Sub ShowInTheTaskbar(hwnd As Long, bShow As Boolean)
    Dim lStyle As Long
    
    ShowWindow hwnd, SW_HIDE
    
    lStyle = GetWindowLong(hwnd, GWL_EXSTYLE)
    
    If bShow = False Then
        If lStyle And WS_EX_APPWINDOW Then
            lStyle = lStyle - WS_EX_APPWINDOW
        End If
    Else
        lStyle = lStyle Or WS_EX_APPWINDOW
    End If
    
    SetWindowLong hwnd, GWL_EXSTYLE, lStyle
    
    'App.TaskVisible = bShow
    
    ShowWindow hwnd, SW_NORMAL
End Sub
```

**Reply by roke-julian-lockhart** [reply]
> https://github.com/twinbasic/twinbasic/discussions/1630#discussioncomment-6493958

Thanks lots for this, @fafalone. I've suggested it at https://github.com/yereverluvinunclebert/Panzer-Earth-gauge-VB6/issues/3#issuecomment-1644860492.
